AppDevAnnouncements/Classes/AnnouncementNetworking.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  AnnouncementNetworking.swift
+//  AppDevAnnouncements
+//
+//  Created by Gonzalo Gonzalez on 11/17/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import FutureNova
+
+public class AnnouncementNetworking {
+
+    private var announcementPath: String
+    private var caller: UIViewController
+
+    private let networking: Networking = URLSession.shared.request
+
+    init(scheme: String, host: String, commonPath: String, announcementPath: String, caller: UIViewController) {
+        Endpoint.setupEndpointConfig(scheme, host, commonPath)
+        self.announcementPath = announcementPath
+        self.caller = caller
+    }
+
+    private func getAnnouncement() -> Future<Response> {
+        return networking(Endpoint.getAnnouncement(announcementPath)).decode()
+    }
+
+    public func retrieveAnnouncement() {
```

Comment:
when is this going to be called?

Reply:
this is called when you want to check to see if there is a backend response and, if there is one, display the notification; otherwise, undefined behavior at the moment besides a console logged error